File: jdroid-java-http-core/src/main/java/com/jdroid/java/http/exception/HttpResponseException.java
package com.jdroid.java.http.exception;


import com.jdroid.java.exception.ErrorCodeException;

/**
 * Exception thrown when there are unknown HTTP errors while communicating with the server.
 */
public class HttpResponseException extends ErrorCodeException {
	
	private static final long serialVersionUID = -1165226174367435109L;
	
	public HttpResponseException(String message) {
		super(HttpErrorCode.HTTP_RESPONSE_ERROR, message);
		setTrackable(true);
	}
	
}

File: jdroid-java-http-core/src/main/java/com/jdroid/java/http/exception/HttpResponseException.kt
package com.jdroid.java.http.exception

import com.jdroid.java.exception.ErrorCodeException

/**
 * Exception thrown when there are unknown HTTP errors while communicating with the server.
 */
class HttpResponseException(message: String) : ErrorCodeException(HttpErrorCode.HTTP_RESPONSE_ERROR, message) {

    init {
        isTrackable = true
    }

    companion object {
        private const val serialVersionUID = -1165226174367435109L
    }
}
